runtime/parachains/src/paras_inherent.rs

```diff
@@ -329,180 +332,190 @@ pub mod pallet {
 			ensure!(!Included::<T>::exists(), Error::<T>::TooManyInclusionInherents);
 			Included::<T>::set(Some(()));
 
-			let ParachainsInherentData {
-				bitfields: mut signed_bitfields,
-				mut backed_candidates,
-				parent_header,
-				mut disputes,
-			} = data;
+			Self::enter_inner::<false>(data)
+		}
+	}
+}
 
-			log::debug!(
-				target: LOG_TARGET,
-				"[enter] bitfields.len(): {}, backed_candidates.len(): {}, disputes.len() {}",
-				signed_bitfields.len(),
-				backed_candidates.len(),
-				disputes.len()
-			);
+impl<T: Config> Pallet<T> {
+	pub(crate) fn enter_inner<const SKIP_CHECKER_CHECKS: bool>(
+		data: ParachainsInherentData<T::Header>,
+	) -> DispatchResultWithPostInfo {
+		let ParachainsInherentData {
+			bitfields: mut signed_bitfields,
+			mut backed_candidates,
+			parent_header,
+			mut disputes,
+		} = data;
 
-			// Check that the submitted parent header indeed corresponds to the previous block hash.
-			let parent_hash = <frame_system::Pallet<T>>::parent_hash();
-			ensure!(
-				parent_header.hash().as_ref() == parent_hash.as_ref(),
-				Error::<T>::InvalidParentHeader,
-			);
+		log::debug!(
+			target: LOG_TARGET,
+			"[enter] bitfields.len(): {}, backed_candidates.len(): {}, disputes.len() {}",
```

Comment:
based on experience, you're going to want the parent header hash here at some point.